utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -117,55 +119,64 @@ impl TrieType for Small {
 /// For more information:
 /// - [ICU Site design doc](http://site.icu-project.org/design/struct/utrie)
 /// - [ICU User Guide section on Properties lookup](https://unicode-org.github.io/icu/userguide/strings/properties.html#lookup)
-pub struct CodePointTrie<'trie, W: ValueWidth, T: TrieType> {
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
```

Comment:
The compiler errors that I'm getting when adding the derive for Serde are very much like the [ones seen recently](https://github.com/unicode-org/icu4x/pull/922#discussion_r714149286) for Zerovec in UnicodeSet. But unlike there, here, there is no custom impl of the Serde traits. Should there be, though, in order to resolve these errors?

Reply:
You shouldn't `derive(Serialize, Deserialize)` on CPT itself, because you need to perform validation on the CPT and have more control over how it is stored in data files.  You should `derive(Serialize, Deserialize)` on data structs, not on exotic types.  See the [style guide](https://github.com/unicode-org/icu4x/blob/main/docs/process/style_guide.md#pre-parsed-fields-exotic-types--suggested): "In most cases, it is insufficient to auto-derive serde::Deserialize on an exotic type. Deserialization must perform data validation in order to retain internal invariants of the exotic type."

Reply:
As discussed: we can now safely `derive(Serialize, Deserialize)` because we eliminated all invariants.